src/ResourceManager/IotHub/Commands.IotHub/help/Add-AzureRmIotHubRoute.md

```diff
@@ -0,0 +1,252 @@
+﻿---
+external help file: Microsoft.Azure.Commands.IotHub.dll-Help.xml
+Module Name: AzureRM.IotHub
+online version: https://docs.microsoft.com/en-us/powershell/module/azurerm.iothub/add-azurermiothubroute
+schema: 2.0.0
+---
+
+# Add-AzureRmIotHubRoute
+
+## SYNOPSIS
+Create a route in IoT Hub
+
+## SYNTAX
+
+### ResourceSet (Default)
+```
+Add-AzureRmIotHubRoute [-ResourceGroupName] <String> [-Name] <String> [-RouteName] <String>
+ [-Source] <PSRoutingSource> [-EndpointName] <String> [-Condition <String>] [-Enabled]
+ [-DefaultProfile <IAzureContextContainer>] [-WhatIf] [-Confirm] [<CommonParameters>]
+```
+
+### InputObjectSet
+```
+Add-AzureRmIotHubRoute [-InputObject] <PSIotHub> [-RouteName] <String> [-Source] <PSRoutingSource>
+ [-EndpointName] <String> [-Condition <String>] [-Enabled] [-DefaultProfile <IAzureContextContainer>] [-WhatIf]
+ [-Confirm] [<CommonParameters>]
+```
+
+### ResourceIdSet
+```
+Add-AzureRmIotHubRoute [-ResourceId] <String> [-RouteName] <String> [-Source] <PSRoutingSource>
+ [-EndpointName] <String> [-Condition <String>] [-Enabled] [-DefaultProfile <IAzureContextContainer>] [-WhatIf]
+ [-Confirm] [<CommonParameters>]
+```
+
+## DESCRIPTION
+Create a route to send specific data source and condition to a desired endpoint.
+
+## EXAMPLES
+
+### Example 1
+```
+PS C:\> Add-AzureRmIotHubRoute -ResourceGroupName "myresourcegroup" -Name "myiothub" -RouteName R1 -Source DeviceMessages -EndpointName E1
+
+RouteName     : R1
+DataSource    : DeviceMessages
+EndpointNames : E1
+Condition     : 
+IsEnabled     : False
+```
+
+Create a new route "R1".
+
+## PARAMETERS
+
+### -Condition
+Condition that is evaluated to apply the routing rule
+
+```yaml
+Type: String
```

Comment:
You need to run your help generation with the `-UseFullTypeName` parameter. It will add the whole typename for every parameter. See:
https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module

Reply:
Done